connectors/opa/main.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+
+	opabl "github.com/ibm/the-mesh-for-data/connectors/opa/lib"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+	"google.golang.org/grpc"
+)
+
+var opaServerURL = ""
+
+const defaultPort = "50082" //synched with opa_connector.yaml
+
+type server struct {
+	pb.UnimplementedPolicyManagerServiceServer
+	opaReader *opabl.OpaReader
+}
+
+func getEnv(key string) string {
+	value, exists := os.LookupEnv(key)
+	if !exists {
+		log.Fatalf("Env Variable %v not defined", key)
+	}
+	log.Printf("Env. variable extracted: %s - %s\n", key, value)
+	return value
+}
+
+func getEnvWithDefault(key string, defaultValue string) string {
+	value, exists := os.LookupEnv(key)
+	if !exists {
+		log.Printf("Env. variable not found, defualt value used: %s - %s\n", key, defaultValue)
+		return defaultValue
+	}
+
+	log.Printf("Env. variable extracted: %s - %s\n", key, value)
+	return value
+}
+
+func (s *server) GetPoliciesDecisions(ctx context.Context, in *pb.ApplicationContext) (*pb.PoliciesDecisions, error) {
+
+	log.Println("Received ApplicationContext")
+	log.Println(in)
+	catalogConnectorAddress := getEnv("CATALOG_CONNECTOR_URL")
+
+	timeOutInSecs := getEnv("CONNECTION_TIMEOUT")
+	timeOut, err := strconv.Atoi(timeOutInSecs)
+
+	if err != nil {
+		log.Printf("Atoi conversion of timeOutinseconds failed: %v", err)
+		return nil, fmt.Errorf("Atoi conversion of timeOutinseconds failed: %v", err)
+	}
+
+	catalogReader := opabl.NewCatalogReader(catalogConnectorAddress, timeOut)
+	eval, err := s.opaReader.GetOPADecisions(in, catalogReader)
+	if err != nil {
+		log.Println("GetOPADecisions err:", err)
+		return nil, err
+	}
+	jsonOutput, _ := json.MarshalIndent(eval, "", "\t")
```

Comment:
Please handle the error here. Thus if there is an error you probably should not log the jsonOutput string.

Reply:
Fixed. Thanks.